src/users/cd-login.vue

```diff
@@ -17,13 +36,52 @@
       return {
         email: '',
         password: '',
+        loginFailed: false,
       };
     },
     methods: {
+      async validateForm() {
+        try {
+          return await this.$validator.validateAll();
+        } catch (e) {
+          return false;
+        }
+      },
       async login() {
-        await UserService.login(this.email, this.password);
-        this.$router.push('/');
+        const valid = await this.validateForm();
+        if (valid) {
+          const response = await UserService.login(this.email, this.password);
+          if (response.body.ok === false) {
+            this.loginFailed = true;
```

Comment:
You can use vee-validate's Errorbag to do backend validation https://github.com/CoderDojo/cp-zen-frontend/pull/108/files#diff-c44cabe05813041a88f869b01b78e0c5R177